static/js/popover_menus_data.js

```diff
@@ -0,0 +1,111 @@
+/* This module provides relevant data to render popovers that require multiple args.
+   This helps keep the popovers code small and keep it focused on rendering side of things. */
+
+import * as feature_flags from "./feature_flags";
+import * as hash_util from "./hash_util";
+import {$t} from "./i18n";
+import * as message_edit from "./message_edit";
+import * as message_lists from "./message_lists";
+import * as muted_users from "./muted_users";
+import * as narrow_state from "./narrow_state";
+import {page_params} from "./page_params";
+import * as stream_data from "./stream_data";
+
+export function get_actions_popover_content_args(message_id) {
```

Comment:
I like this pattern! I think `get_actions_popover_content_context` might be a better naming scheme for these -- we're generating a rendering context to pass into the template.

Reply:
Okay, done!